### DISCUSSION TITLE
How to determine which observable change caused component to re-render?

### DISCUSSION THREAD

**Discussion by pavelustenko** [mobxjs/mobx#3768] [Q&A]
Hello.
I'm new with mobx, but I delighted about it.
I wonder if there is some utility to track re-renders and changes in observables that caused them, alike discussed https://stackoverflow.com/questions/41004631/trace-why-a-react-component-is-re-rendering

**Comment by mweststrate** [comment]
https://mobx.js.org/analyzing-reactivity.html

**Reply by domehead100** [reply]
Yes, trace, spy, intercept are incredibly useful at times.  Not sure why they are being removed in Mobx v7 (per the roadmap issue).

Another manual technique that I’ve used in the past is to use Object.defineProperty to change the property descriptor, such as making it not writeable, which would cause an error when it is changed, and then you know the line of code that changed it.
